From 196c5a01c32d875843e975b9642f242275020e41 Mon Sep 17 00:00:00 2001
From: Veaceslav Falico <vfalico@redhat.com>
Date: Thu, 21 Aug 2014 11:28:31 -0400
Subject: [ethernet] netxen: remove unnecessary pci_set_drvdata()

Message-id: <1408620515-703-7-git-send-email-vfalico@redhat.com>
Patchwork-id: 88513
O-Subject: [PATCH RHEL7 06/10] net: netxen: remove unnecessary pci_set_drvdata()
Bugzilla: 1089356
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

RHBZ#1089356

Testing:
Locally and by QLogic (which supplied the patch)

Description:
Backport of the following commit:

commit b23eb795267fb32113c204870ecfadba2a7ecf7c
Author: Jingoo Han <jg1.han@samsung.com>
Date:   Tue Oct 22 13:15:09 2013 +0900

    net: netxen: remove unnecessary pci_set_drvdata()

    The driver core clears the driver data to NULL after device_release
    or on probe failure. Thus, it is not needed to manually clear the
    device driver data to NULL.

    Signed-off-by: Jingoo Han <jg1.han@samsung.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/netxen/netxen_nic_main.c b/drivers/net/ethernet/qlogic/netxen/netxen_nic_main.c
index 7460d58..b1ef285 100644
--- a/drivers/net/ethernet/qlogic/netxen/netxen_nic_main.c
+++ b/drivers/net/ethernet/qlogic/netxen/netxen_nic_main.c
@@ -1630,7 +1630,6 @@ err_out_free_res:
  pci_release_regions(pdev);
 
 err_out_disable_pdev:
- pci_set_drvdata(pdev, NULL);
  pci_disable_device(pdev);
  return err;
 }
@@ -1689,7 +1688,6 @@ static void netxen_nic_remove(struct pci_dev *pdev)
 
  pci_release_regions(pdev);
  pci_disable_device(pdev);
- pci_set_drvdata(pdev, NULL);
 
  free_netdev(netdev);
 }
-- 
1.7.1